packages/handlersjs-http/lib/handlers/error.handler.ts

```diff
@@ -72,18 +69,12 @@ export class ErrorHandler implements HttpHandler {
 
   handle(context: HttpHandlerContext): Observable<HttpHandlerResponse>{
 
-    if (!context) {
-
-      this.logger.verbose('Context was not provided');
-
-      return throwError(() => new Error('A context must be provided'));
-
-    }
+    context.logger.setLabel(this);
 
     return this.nestedHandler.handle(context).pipe(
       catchError((error) => {
 
-        this.logger.error('Error occurred: ', error);
+        context.logger.error('Error occurred: ', error);
```

Comment:
Just a question, not part of a broader review. What information is passed here when logging the whole error object? It should include the stack trace, so we should add that if it's not present by default. That will help us figure out where the error was caused, since this handler is catching unhandled errors thrown by other functions and handlers.

Reply:
I'd assume that `error` is an `Error` instance. (https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/error#instance_properties)

Reply:
Can we test this easily? Just incase it's not what we expect, then we can update it as part of this PR

Reply:
Example of `logger.log('bla', error)`:
```
[2023-04-07T13:05:09.523Z NodeHttpRequestResponseHandler] Error: {
  requestId: 'cd73eda9-21e8-4bce-a21c-bce668620cf3',
  correlationId: '0530a5f8-513b-49c9-8ca9-5ab2d6109b2d'
} Error: test error
    at MockHttpHandler.handle (/Users/arthur/Documents/handlersjs/packages/handlersjs-http/dist/handlers/http.handler.mock.js:23:15)
    at RoutedHttpRequestHandler.handle (/Users/arthur/Documents/handlersjs/packages/handlersjs-http/dist/handlers/routed-http-request.handler.js:83:40)
    at /Users/arthur/Documents/handlersjs/packages/handlersjs-http/dist/servers/node/node-http-request-response.handler.js:112:37
    at /Users/arthur/Documents/handlersjs/packages/handlersjs-http/node_modules/rxjs/dist/cjs/internal/operators/switchMap.js:17:35
    at OperatorSubscriber._this._next (/Users/arthur/Documents/handlersjs/packages/handlersjs-http/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:33:21)
    at OperatorSubscriber.Subscriber.next (/Users/arthur/Documents/handlersjs/packages/handlersjs-http/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
    at /Users/arthur/Documents/handlersjs/packages/handlersjs-http/node_modules/rxjs/dist/cjs/internal/operators/map.js:10:24
    at OperatorSubscriber._this._next (/Users/arthur/Documents/handlersjs/packages/handlersjs-http/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:33:21)
    at OperatorSubscriber.Subscriber.next (/Users/arthur/Documents/handlersjs/packages/handlersjs-http/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
    at /Users/arthur/Documents/handlersjs/packages/handlersjs-http/node_modules/rxjs/dist/cjs/internal/operators/map.js:10:24
    ```